hackathon_site/event/test_api.py

```diff
@@ -90,6 +90,102 @@ def test_user_has_profile(self):
         self.assertEqual(response.json(), serializer.data)
 
 
+class JoinTeamTestCase(SetupUserMixin, APITestCase):
+    def setUp(self):
+        super().setUp()
+        self.group = Group.objects.create(name="Test Users")
+        self.user.groups.add(self.group)
+        self.team = Team.objects.create()
+
+        self.profile = Profile.objects.create(user=self.user, team=self.team)
+        self.team_code = self.team.team_code
+        self.view_name = "api:event:join-team"
+
+    def _build_view(self, team_code):
+        return reverse(self.view_name, kwargs={"team_code": team_code})
+
+    def test_join_and_delete(self):
+        """
+        When a member wants to join a team, if their current team only includes
+        them, their current team is deleted.
+        """
+        self._login()
+        self.client.post(self._build_view(self.team.team_code))
```

Comment:
Instead of using a post request to put the user on a team the first time, can you just set it up directly from a model query? Helps keep this test isolated in what it is testing (that leaving an empty team deletes it)